vllm/config.py

```diff
@@ -3400,19 +3408,21 @@ def __post_init__(self):
 
         if self.compilation_config is None:
             self.compilation_config = CompilationConfig()
-        if envs.VLLM_USE_V1 and self.model_config is not None and \
+        if self.use_v1 and self.model_config is not None and \
             not self.model_config.enforce_eager:
             # NOTE(woosuk): Currently, we use inductor because the piecewise
             # CUDA graphs do not work properly with the custom CUDA kernels.
             # FIXME(woosuk): Disable inductor to reduce the compilation time
             # and avoid any potential issues with the inductor.
+
             self.compilation_config.custom_ops = ["none"]
             self.compilation_config.use_cudagraph = True
             self.compilation_config.use_inductor = True
             self.compilation_config.cudagraph_num_of_warmups = 1
             self.compilation_config.pass_config.enable_fusion = False
             self.compilation_config.pass_config.enable_reshape = False
             self.compilation_config.level = CompilationLevel.PIECEWISE
+            self.compilation_config.set_splitting_ops_for_v1()
```

Comment:
This feels like it should be a method `CompilationConfig.enable_v1()`. I think it'd also be good to rethink if `enforce_eager` makes sense as part of `model_config`. Because this forcibly overrides compilation config in a way that makes it impossible to control from the outside.

Reply:
yeah good idea

Reply:
Separately @ProExpertProg - does this mean that we are not using our fusions in V1?

Reply:
Yes, and I think it also means we can't even enable them because the config is hard coded here.

Reply:
We should address this all in a follow up. This PR is too big